Kinvey.Core/Store/PushRequest.cs

```diff
@@ -15,180 +15,360 @@
-				entity = await request.ExecuteAsync();
+                NetworkRequest<T> request = Client.NetworkFactory.buildUpdateRequest<T>(pwa.collection, localEntity, pwa.entityId);
+                entity = await request.ExecuteAsync();
 
                 Cache.UpdateCacheSave(entity, pwa.entityId);
 
                 result = SyncQueue.Remove(pwa);
 
-				if (result == 0)
-				{
-					offset++;
-				}
-			}
-			catch (KinveyException ke)
-			{
-				response.AddKinveyException(ke);
-				offset++;
-			}
-
-			return entity;
-		}
-
-		private async Task<T> HandlePushDELETE(PendingWriteAction pwa)
-		{
-			try
-			{
-				int result = 0;
-
-				NetworkRequest<KinveyDeleteResponse> request = Client.NetworkFactory.buildDeleteRequest<KinveyDeleteResponse>(pwa.collection, pwa.entityId);
-				KinveyDeleteResponse kdr = await request.ExecuteAsync();
-
-				if (kdr.count == 1)
-				{
-					result = SyncQueue.Remove(pwa);
-
-					if (result == 0)
-					{
-						offset++;
-					}
-				}
-			}
-			catch (KinveyException ke)
-			{
-				response.AddKinveyException(ke);
-				offset++;
-			}
-
-			return default(T);
-		}
-	}
+                if (result == 0)
+                {
+                    offset++;
+                }
+            }
+            catch (KinveyException ke)
+            {
+                kinveyException = ke;
+                offset++;
+            }
+
+            return new Tuple<T, KinveyException, int>(entity, kinveyException, offset);
+        }
+
+        private async Task<Tuple<T, KinveyException, int>> HandlePushDELETE(PendingWriteAction pwa)
+        {
+            var offset = 0;
+            KinveyException kinveyException = null;
+
+            try
+            {
+                int result = 0;
+
+                NetworkRequest<KinveyDeleteResponse> request = Client.NetworkFactory.buildDeleteRequest<KinveyDeleteResponse>(pwa.collection, pwa.entityId);
+                KinveyDeleteResponse kdr = await request.ExecuteAsync();
+
+                if (kdr.count == 1)
+                {
+                    result = SyncQueue.Remove(pwa);
+
+                    if (result == 0)
+                    {
+                        offset++;
+                    }
+                }
+            }
+            catch (KinveyException ke)
+            {
+                kinveyException = ke;
+                offset++;
+            }
+
+            return new Tuple<T, KinveyException, int>(default(T), kinveyException, offset);
+        }
+
+        #endregion Single actions pushes
+
+        #region Multiple actions pushes
+
+        private async Task<PushDataStoreResponse<T>> PushMultiPostActionsAsync()
+        {
+            var response = new PushDataStoreResponse<T>();
+            var limit = 10 * Constants.NUMBER_LIMIT_OF_ENTITIES;
+            var offset = 0;
+            var pendingPostActions = SyncQueue.GetFirstN(limit, offset, "POST");
```

Comment:
shouldn't this call be made without a limit? or maybe do a count first so you can limit to `100` and go page by page?

Reply:
@heyzooi 
You suggest fetching all data from the local database one time and putting it in a local variable. After this according to your idea we will go page by page. It will help us avoid often interaction with the local database. Am I right?

Reply:
right, because I thought the query would lazy load the records, but if it's not your current code is better and way more efficient. thanks for the explanation!